models/models.js

```diff
@@ -1,9 +1,20 @@
 const db = require("../db/connection");
 
-exports.getTopicsMod = () => {
+exports.fetchTopicsMod = () => {
   return db.query(`SELECT * FROM topics;`).then((topics) => {
-    console.log(topics.rows);
-
     return topics.rows;
   });
 };
+
+exports.fetchArticleMod = (article_id) => {
+  return db
+    .query(
+      `SELECT * FROM articles JOIN users ON articles.author = users.username`
+    )
+    .then((article) => {
+      if (article_id > article.rows.length) {
+        return Promise.reject({ status: 404, msg: "article does not exist" });
+      }
+      return article.rows[article_id - 1];
+    });
+};
```

Comment:
A couple of points to note here:
- We only want to `JOIN` tables together when we need to display/use the data from some other table. In this case, all we want do at the moment is get info about a particular article - is it necessary to join it onto the `users` table?
- Imagine if there were 1,000,000 articles in our database (for a website like reddit, for example). Currently this model would fetch all 1,000,000 of them just to send info about 1 of them back to the client. Perhaps there is a way to tell SQL to only give us the article we want?